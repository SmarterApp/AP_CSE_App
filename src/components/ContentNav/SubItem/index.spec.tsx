import React from 'react';
import { mount } from 'enzyme';
import { SubItem } from '.';

describe('SubItem', () => {
  it('should be active', () => {
    const subItemActive = mount(<SubItem name={'Test Name'} active={true} />);

    subItemActive.simulate('click');
    expect(subItemActive).toMatchSnapshot();

    subItemActive.unmount();
  });
});
