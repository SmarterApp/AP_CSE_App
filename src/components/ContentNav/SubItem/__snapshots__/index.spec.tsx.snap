// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubItem should be active 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Unknown
    active={true}
    name="Test Name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "name": "Test Name",
    },
    "ref": null,
    "rendered": Object {
      "instance": <li
        class="active"
        role="menuitem"
      >
        <p />
        <style>
          
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      
        </style>
      </li>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <p />,
          <style
            jsx={true}
          >
            
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      
          </style>,
        ],
        "className": "active",
        "onClick": [Function],
        "role": "menuitem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <p />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": <style>
            
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      
          </style>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      ",
            "jsx": true,
          },
          "ref": null,
          "rendered": Array [
            "
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      ",
          ],
          "type": "style",
        },
      ],
      "type": "li",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "name": "Test Name",
      },
      "ref": null,
      "rendered": Object {
        "instance": <li
          class="active"
          role="menuitem"
        >
          <p />
          <style>
            
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      
          </style>
        </li>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <p />,
            <style
              jsx={true}
            >
              
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      
            </style>,
          ],
          "className": "active",
          "onClick": [Function],
          "role": "menuitem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <p />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": <style>
              
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      
            </style>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      ",
              "jsx": true,
            },
            "ref": null,
            "rendered": Array [
              "
        * {
          font-family: PT Serif;
        }
        li {
          display: flex;
          align-items: center;
          color: #63666A;
          font-size: 14px;
        }
        p {
          margin: 5px 2em 5px 1em;
        }
        .active {
          color: #006298;
        }
      ",
            ],
            "type": "style",
          },
        ],
        "type": "li",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
